src/client/dfs/dfs.c

```diff
@@ -2255,15 +2358,33 @@ dfs_lookup_rel(dfs_t *dfs, dfs_obj_t *parent, const char *name, int flags,
 			stbuf->st_size = size;
 			stbuf->st_blocks = (stbuf->st_size + (1 << 9) - 1) >> 9;
 		}
-		break;
 	case S_IFLNK:
-		/* Create a truncated version of the string */
-		D_STRNDUP(obj->value, entry.value, entry.value_len + 1);
-		if (obj->value == NULL)
-			D_GOTO(err_obj, rc = ENOMEM);
-		D_FREE(entry.value);
-		if (stbuf)
-			stbuf->st_size = entry.value_len;
+		if (flags & O_NOFOLLOW) {
+			/* Create a truncated version of the string */
+			D_STRNDUP(obj->value, entry.value, entry.value_len + 1);
```

Comment:
Seems the segfault is because entry.value is NULL at this point.

Reply:
Thanks @ashleypittman . It looks like I messed up the merge here. There should be a break statement before the S_IFLNK case. The S_IFREG is falling through here

Reply:
I saw you'd retriggerd the Unit Test phase and I didn't think that was going to work.

Reply:
For some reason, when I looked at the logs, I was seeing some erroneous error, so I just restarted. But I think maybe I had just clicked the wrong link :)